src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,47 @@
+package duke;
+
+import java.io.Serializable;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * This class represents a Deadline task.
+ */
+public class Deadline extends Task implements Serializable {
+    private LocalDateTime by;
+    private static final DateTimeFormatter dateFormatter =
+            DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+
+    /**
+     * Creates a deadline task.
+     * @param description indicates what the task is about
+     *                    and when the deadline is
+     * @throws DukeException if input does not meet required
+     * format or '/by' is not specified.
+     */
+    public Deadline(String description) throws DukeException {
+        super(description);
+        String[] splitDescription = description.split(" /by ", 2);
+
+        try {
+            this.description = splitDescription[0];
+            this.by = LocalDateTime.parse(splitDescription[1], dateFormatter);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new DukeException("Please enter a deadline using \"/by\".");
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Please enter a date with the format "
+                    + "dd/MM/yyyy HHmm.\n" + e.getMessage());
+        }
+    }
+
+    /**
+     * Returns a string representation of the Deadline Task.
+     * @return a string representation of the deadline Task.
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString()
+                + " (by: " + by.format(dateFormatter) + ")";
```

Comment:
Alternatively, you can format your dates in the constructor first and use it for future methods.

Reply:
I implemented this actually to keep the date stored as a LocalDateTime object and only convert it to a String using this formatter when actually needed. I guess storing a String representation could also work. Thank you for the suggestion!